status-im/status-legacy
# RAM bundle

1. I haven't found the way to create a release bundle with cljs compiled with advanced optimizations. It just fails with meaningless exception. The reason why I didn't realize that from the beginning is that our PR builds are not bundled as releases. I was under impression that they are. 
2. Thus there is no impressive boost because of this change. The big part of it was minimized cljs build. So now it is only **14%** of js loading, and even less of the whole startup. 
3. I will try to figure out how to integrate cljs with advanced optimization in a separate task. It works when js is bundled for debugging, so it still might be possible with release version.
4. Some modules still might be stripped out of initial bundle, but this doesn't really matter at the moment because bundle is **11,697,115B** where **11,097,753B** is cljs. So the first thing to minimize here is cljs. (release is **13,956,525B**)

## What's done in this PR
- almost all modules (except "react-native", "realm" and few more) are required only by request, not on app initialisation 
- RAM bundle is enabled on android

**testing notes**:
App is bundled as a release, this should be reverted before merging as we build PR differently at the moment.
In order to compare startup time:
- install the last release version and measure time necessary for loading intro screen
- remove app
- install this PR's version
- measure startup time again

status: ready
